### PR TITLE
fix: remove TLS destruction

### PR DESCRIPTION
**Pull request by richard-townsend-arm**
#### Description of Change

Building with `dchecks_always_on=true` in release configuration seems to introduce a crash when content processes exit because the TLS is double-freed, which shows up in the Windows on Arm unit tests as an incorrect exit code. Removing the TLS destruction code (which is only needed in component builds) resolves the flakiness.

This issue affects master, but also affects 6-0-x.

CC: @jkleinsc @MarshallOfSound 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: no-notes

### CODE REVIEW

**Comment by richard-townsend-arm** [comment]
CI seems to have flaked due to Github downtime.

**Comment by deepak1556** [comment]
I don't have much context on this original patch, /cc @zcbenz who wrote it. But CRT is no longer a DLL in release configuration since electron 4.x and this patch specifically is for the case where its a DLL, not sure if we still need to maintain it.

**Comment by richard-townsend-arm** [comment]
So if you build a debug build, you won't get a runtime CRT dll anymore, so therefore we can just remove this whole TLS-freeing thing?

**Comment by nornagon** [comment]
hmm, i think you might still get a runtime crt dll in debug mode. but in testing/release we build with static crt. see https://chromium.googlesource.com/chromium/src/+/76.0.3809.20/build/config/win/BUILD.gn#386

**Comment by richard-townsend-arm** [comment]
OK: I strongly reckon that if this workaround isn't required (so long as nobody cares about debug builds) then I think it's probably best to remove guarded section. I'll put it into testing this evening and hopefully have a new patch out by tomorrow.

File: docs/development/testing.md [hunk 1]
```diff
@@ -48,3 +48,12 @@ would run `npm run test -- -g ipc`.
...
 1. Select _Japanese Supplemental Fonts_ and click _Install_.
+
+
+Some tests which rely on precise pixel measurements may not work correctly on
+devices with Hi-DPI screen settings due to floating point precision errors.
+To run these tests correctly, make sure the device is set to 100% scaling.
+
+To configure display scaling:
+1. Push the Windows key and search for _Display settings_.
+1. Under _Scale and layout_, make sure that the device is set to 100%.
```
Comment (nornagon) [inline docs/development/testing.md]: this seems unrelated?

Comment (richard-townsend-arm) [reply docs/development/testing.md]: Apologies, it looks like my tree was not quite clean.

**Comment by richard-townsend-arm** [comment]
May need to retry [`mas-testing-tests`](https://circleci.com/gh/electron/electron/270355?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link), seems to have this in the log:
```
[1233:0723/110648.546693:WARNING:ca_layer_tree_coordinator.mm(55)] Blank frame: No overlays or CALayers
Received signal 4 <unknown> 00011500ebc4
0   Electron Framework                  0x00000001192b12e9 base::debug::CollectStackTrace(void**, unsigned long) + 9
1   Electron Framework                  0x0000000119164b63 base::debug::StackTrace::StackTrace() + 19
2   Electron Framework                  0x00000001192b1181 base::debug::(anonymous namespace)::StackDumpSignalHandler(int, __siginfo*, void*) + 2385
3   libsystem_platform.dylib            0x00007fff70249f5a _sigtramp + 26
4   ???                                 0x0000000000000030 0x0 + 48
5   Electron Framework                  0x0000000114ea49bd mate::internal::Dispatcher<void ()>::DispatchToCallback(v8::FunctionCallbackInfo<v8::Value> const&) + 173
6   Electron Framework                  0x0000000116a699ae v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) + 1118
7   Electron Framework                  0x0000000116a01141 v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) + 1137
8   Electron Framework                  0x00000001169ff53a v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) + 810
9   Electron Framework                  0x00000001169fef8c v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) + 140
10  Electron Framework                  0x0000000117941fc0 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit + 64
11  Electron Framework                  0x00000001176e54d5 Builtins_InterpreterEntryTrampoline + 949
12  Electron Framework                  0x00000001176db57d Builtins_JSEntryTrampoline + 93
[end of stack trace]
```

**Comment by deepak1556** [comment]
That renderer crash is not related to this change, its been present for some versions now.

**Comment by richard-townsend-arm** [comment]
Cool, @nornagon: are you happy with this solution?

**Comment by jkleinsc** [comment]
Merging as mac failure is known issue unrelated to this PR.
